File: Python_Practice/LeetCode/0344 Reverse String/0344-Reverse-String.py
class Solution(object):
    def reverseString(self, s):
        """
        :type s: List[str]
        :rtype: None Do not return anything, modify s in-place instead.
        """
        n = len(s)
        idx = 0
        temp = s[0]
        for i in range(n-1, n/2 - 1, -1):
            #print("i = ", i, ", idx =", idx)
            #print("s[i] = ", s[i], ", s[idx] = ", s[idx])
            temp = s[i]
            s[i] = s[idx]
            s[idx] = temp
            idx += 1
            #print(s)

File: C++_Practice/LeetCode/0344 Reverse String/0344-Reverse-String.cpp
class Solution {
public:
    void reverseString(vector<char>& s) {
        int n = s.size();
        char temp;
        int idx = 0;
        for (int i = n-1; i >= n/2; i--){
            temp = s[i];
            s[i] = s[idx];
            s[idx] = temp;
            idx++;
        }
    }
};